src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int indexOfLastAddedElement = 0;
+    private Object[] array = new Object[10];
+
     @Override
     public void add(T value) {
-
+        if (indexOfLastAddedElement >= array.length) {
+            array = Arrays.copyOf(array, (int) (array.length * 1.5));
```

Comment:
[You need to use `System.arraycopy()`](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-systemarraycopy-to-move-your-array-elements). Fix it in all places
extract logic that inside if to `grow()` method. Reuse it everywhere

Reply:
not fixed, `Arrays.copyOf` is not allowed